pkg/pillar/agentlog/agentlog.go

```diff
@@ -341,14 +340,15 @@ func roundToMb(b uint64) uint64 {
 	return mb
 }
 
-func Init(agentName string, curpart string) error {
+func Init(agentName string) {
+	curpart := EveCurrentPartition()
 	if curpart != "" {
 		zboot.SetCurpart(curpart)
 	}
```

Comment:
Does this really belong here? Init is called by every agent. Curpart should be set by 1 agent ..

Ideally, device-steps or some other agent that starts up first should sent this.

Reply:
https://www.pivotaltracker.com/story/show/171869555
I think the story description will answer your questions.

Reply:
We don't have to call zboot.SerCurpart(). That is just an optimization.

Reply:
Will remove this along with some other code that I believe is dead.

Reply:
I think we should remove the call to zboot.SerCurpart() from here. Where is this actually supposed to be set?

Reply:
+1 to removing if curpart != "" {... block -- I see no valuable use for it here anymore